package.json

```diff
@@ -9,15 +9,15 @@
   "license": "MIT",
   "scripts": {
     "rebuild": "lerna run --stream rebuild",
-    "coverage": "nyc report --reporter=text-lcov > coverage.lcov && codecov",
     "format": "eslint packages test --fix",
     "generate": "plop",
     "lint": "eslint packages test && lerna run --stream lint",
     "outdated": "lerna exec --no-bail npm outdated && npm outdated",
     "postinstall": "lerna bootstrap --no-ci",
     "prebuild": "lerna run --stream prebuild",
     "publish": "lerna publish --exact",
-    "test": "nyc --reporter=html --reporter=text --reporter lcovonly mocha",
+    "coverage:publish:node": "codecov --disable=gcov --file=coverage/lcov.info",
```

Comment:
IMHO, this is the main part that needs some workshopping, in conjunction with #2314. Each OS has slightly different steps for collecting coverage reports and publishing the data to codecov.

Previously, `scripts#coverage` reran the tests to generate the raw coverage data a second time, before uploading to codecov, to ensure the coverage files were all generated. I felt this was redundant with just running the `test` script first.

As for why I picked this script name, it lines up with names chosen in #2314. But, again, I'm rather open to these names changing.

Thinking about this more, codecov *should* be able to hand some of the cross-platform issues. I'll look at that to see if I can simplify the coverage steps.

Reply:
Currently, the [CI action runs](https://github.com/serialport/node-serialport/blob/master/.github/workflows/test.yml#L31-L32):

```bash
npm test
npm run coverage
```

Which, respectively, just run:

```bash
nyc --reporter=html --reporter=text --reporter lcovonly mocha
nyc report --reporter=text-lcov > coverage.lcov && codecov
```

While the `nyc report --reporter=text-lcov > coverage.lcov` part generates `coverage.lcov`, `codecov` doesn't need it. If you look at the [CI logs](https://github.com/serialport/node-serialport/runs/4330567254?check_suite_focus=true), you'll see that `codecov` finds two identical coverage files:

```log
==> Scanning for reports
    + /home/runner/work/node-serialport/node-serialport/coverage/lcov.info
    + /home/runner/work/node-serialport/node-serialport/coverage.lcov
```

`coverage/lcov.info` is generated by the `nyc ... mocha` with the `lcovonly` "reporter" enabled and has the exact same information as `text-lcov`. So the `nyc report ... > coverage.lcov` is a little redundant.

Looking at this more closely, I think I may have previously missed the `report` command and just assumed it was fully rerunning the tests. This may have been incorrect.

[Looking at the docs for `report`](https://www.npmjs.com/package/nyc#combining-reports-from-multiple-runs), it seems the intent is to use `report` in more complicated testing scenarios where a few tests are run ahead of time and the final reports are generated as a final step. This could be valuable but currently `codecov` also handles merging coverage information from multiple related runs so it doesn't seem as necessary to me.